strs.go

```diff
@@ -57,6 +57,41 @@ func (db *RoseDB) MGet(keys [][]byte) ([][]byte, error) {
 	return values, nil
 }
 
+// GetRange returns the substring of the string value stored at key,
+// determined by the offsets start and end.
+func (db *RoseDB) GetRange(key []byte, start, end int) ([]byte, error) {
+	db.strIndex.mu.RLock()
+	defer db.strIndex.mu.RUnlock()
+
+	val, err := db.getVal(db.strIndex.idxTree, key, String)
+	if err != nil {
+		return nil, err
+	}
+	// Negative offsets can be used in order to provide an offset starting from the end of the string.
+	// So -1 means the last character, -2 the penultimate and so forth
+	if start < 0 {
+		start = len(val) + start
+		if start < 0 {
+			start = 0
+		}
+	}
+	if end < 0 {
+		end = len(val) + end
+		if end < 0 {
+			end = 0
+		}
+	}
+
+	// handles out of range requests by limiting the resulting range to the actual length of the string.
+	if start > len(val)-1 {
+		return []byte{}, nil
+	}
+	if end > len(val)-1 {
+		end = len(val) - 1
+	}
+	return val[start : end+1], nil
```

Comment:
If start and end both are positive numbers, and start > end. eg:
val is "abcde", start = 3, end = 1

val[start:end] will panic

Reply:
Oh, I did it.
I had completely misplaced that case.
Thanks for pointing that out!